src/memory.rs

```diff
@@ -184,6 +279,88 @@ impl Mul<f32> for MemoryQuantity {
     }
 }
 
+impl Div<f32> for MemoryQuantity {
+    type Output = Self;
+
+    fn div(self, rhs: f32) -> Self::Output {
+        self * (1. / rhs)
+    }
+}
+
+impl Sub<MemoryQuantity> for MemoryQuantity {
+    type Output = MemoryQuantity;
+
+    fn sub(self, rhs: MemoryQuantity) -> Self::Output {
+        if rhs.unit == self.unit {
+            MemoryQuantity {
+                value: self.value - rhs.value,
+                unit: self.unit,
+            }
+        } else if rhs.unit < self.unit {
+            MemoryQuantity {
+                value: self.scale_to(rhs.unit).value - rhs.value,
+                unit: rhs.unit,
+            }
+        } else {
+            MemoryQuantity {
+                value: self.value - rhs.scale_to(self.unit).value,
+                unit: self.unit,
+            }
+        }
+    }
+}
+
+impl Add<MemoryQuantity> for MemoryQuantity {
+    type Output = MemoryQuantity;
+
+    fn add(self, rhs: MemoryQuantity) -> Self::Output {
+        if rhs.unit == self.unit {
+            MemoryQuantity {
+                value: self.value + rhs.value,
+                unit: self.unit,
+            }
+        } else if rhs.unit < self.unit {
+            MemoryQuantity {
+                value: self.scale_to(rhs.unit).value + rhs.value,
+                unit: rhs.unit,
+            }
+        } else {
+            MemoryQuantity {
+                value: self.value + rhs.scale_to(self.unit).value,
+                unit: self.unit,
+            }
+        }
+    }
+}
+
+impl PartialOrd for MemoryQuantity {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        let this_val = self.scale_to(BinaryMultiple::get_smallest()).value;
+        let other_val = other.scale_to(BinaryMultiple::get_smallest()).value;
+        this_val.partial_cmp(&other_val)
+    }
+}
+
+impl Ord for MemoryQuantity {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        let this_val = self.scale_to(BinaryMultiple::get_smallest()).value;
+        let other_val = other.scale_to(BinaryMultiple::get_smallest()).value;
+        // Note: We just assume that our values are always not NaN, so we are actually Ord.
+        // A MemoryQuantity with NaN is not permissible.
```

Comment:
Im not a fan of having unwraps. This is 100% sure :D ?

Reply:
Of course you could create a MemoryQuantity with a `NaN` value, and this will break. We could try to guard against this at all the places where the MemoryQuantity is created, and try to not have a NaN inside like that to make this unwrap safer, but we'd still have to have the unwrap here.

The problem is that floats are actually not ordered, because of NaN. so in the `None` case, what should we return? There is only "Less", "Equal", or "Greater". We can't return an error, because this is a trait function and we _have_ to return an Ordering. What do you think is the best way forward here?

Reply:
Alright, I see now that an unwrap is not good. This is unfortunate, but I've opted to simply remove the `Ord` implementation.

Reply:
fine with me :)